modules/mono/editor/hostfxr_resolver.cpp

```diff
@@ -198,7 +198,13 @@ bool get_default_installation_dir(String &r_dotnet_root) {
 	r_dotnet_root = path::join(program_files_dir, "dotnet");
 	return true;
 #elif defined(MACOS_ENABLED)
-	r_dotnet_root = "/usr/local/share/dotnet";
+	if (DirAccess::exists("/opt/homebrew/opt/dotnet/libexec/host/fxr")) {
+		r_dotnet_root = "/opt/homebrew/opt/dotnet/libexec";
+	} else if (DirAccess::exists("/usr/local/opt/dotnet/libexec/host/fxr")) {
+		r_dotnet_root = "/usr/local/opt/dotnet/libexec";
+	} else if (DirAccess::exists("/usr/local/share/dotnet/host/fxr")) {
+		r_dotnet_root = "/usr/local/share/dotnet";
+	}
```

Comment:
AFAIK, the `hostfxr_resolver` is supposed to mirror the [`fxr_resolver`](https://github.com/dotnet/runtime/blob/6e422d6f6eb6bac07ea066d06fe42856459a5d79/src/native/corehost/fxr_resolver.cpp) from .NET so I wouldn't add these here.

The .NET implementation seems to only return `/usr/local/share/dotnet`:

https://github.com/dotnet/runtime/blob/6e422d6f6eb6bac07ea066d06fe42856459a5d79/src/native/corehost/hostmisc/pal.unix.cpp#L581-L594

Reply:
Without this, this won't actually help anything, as it still won't be using the homebrew installed dotnet as the editor runtime.

Reply:
I was just wondering if it should be added somewhere else. But IIRC we try to find hostfxr from the `dotnet` found in PATH, is `dotnet` not in the PATH when installed using homebrew?

Reply:
macOS apps ignore all user defined environment variables, `PATH` is always `/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin` (only `/usr/local/bin` is writable).